src/main/java/com/nha/abdm/wrapper/hip/hrp/dataTransfer/HealthInformationService.java

```diff
@@ -124,12 +124,44 @@ public void healthInformation(HIPHealthInformationRequest hipHealthInformationRe
     log.debug(
         "health information acknowledgment request body : "
             + onHealthInformationRequest.toString());
+    // Acknowledge to gateway that health information request has been received.
     healthInformationAcknowledgementRequest(
         hipHealthInformationRequest, onHealthInformationRequest);
+    // Prepare health information bundle request which needs to be sent to HIP.
     HealthInformationBundle healthInformationBundle =
-        fetchHealthInformationBundle(hipHealthInformationRequest);
-    pushHealthInformation(healthInformationBundle, consentId);
-    healthInformationPushAcknowledge(hipHealthInformationRequest, consentId);
+        fetchHealthInformationBundle(hipHealthInformationRequest, gatewayRequestId);
```

Comment:
We might have to validate if the consent has status in GRANTED state, and also check for the date range etc.